<?xml version="1.0"?>
<launch>
  <rosparam name="use_sim_time" value="true"/>
  <arg name="use_sim_time" default="true" />
  <arg name="gui" default="true" />
  <arg name="headless" default="false" />
  <arg name="map_file" default="$(find navigation)/maps/FHWS-first-floor/fhws_two_lidar_loop_08_karto_kind_of_loop_with_bigger_buffer_perfekt_map.yaml"/>


  <!-- Map server -->
  <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)"/>
   
  <include file="$(find amcl_localization)/launch/amcl_launch.launch"/>

  <!-- prevent backwards movement if not necessary -->
  <node pkg="navigation" type="my_planner" name="my_planner" >
        <rosparam file="$(find navigation)/param/my_planner.yaml" command="load"/>
  </node>

  <!-- move base Pathplaning -->
  <node pkg="move_base" type="move_base" respawn="false" name="move_base" output="screen">
    # Global planner
    <param name="base_global_planner" value="global_planner/GlobalPlanner" />  
    
    # Local planner
    <param name="base_local_planner" value="dwa_local_planner/DWAPlannerROS" /> 
    
    # Costmap
    <rosparam file="$(find navigation)/param/costmap_common_params_wheelchair.yaml" command="load" ns="global_costmap" />
    <rosparam file="$(find navigation)/param/costmap_common_params_wheelchair.yaml" command="load" ns="local_costmap" />
    <rosparam file="$(find navigation)/param/local_costmap_params.yaml" command="load" />
    <rosparam file="$(find navigation)/param/global_costmap_params.yaml" command="load" />
    
    # Movebase
    <rosparam file="$(find navigation)/param/move_base_params.yaml" command="load" />
    
    # global planer parameter
    <rosparam file="$(find navigation)/param/global_planner/global_global_planner_params.yaml" command="load"/>
    
    # local planer parameter
    <rosparam file="$(find navigation)/param/local_planner/dwa_local_planner_params_wheelchair.yaml" command="load" />
    
    <remap from="cmd_vel" to="/cmd_vel"/>
    <remap from="odom" to="odom"/>
  </node>

  <node pkg="rviz" type="rviz" name="rviz" required="true"  args="-d $(find navigation)/rviz/rviz_config.rviz"/>


</launch>
